File: mobile_apps/ALADDIN_Android/app/src/main/java/family/aladdin/android/ui/components/cards/FunctionCard.kt
package family.aladdin.android.ui.components.cards

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import family.aladdin.android.ui.theme.*

/**
 * 📇 Function Card
 * Карточка функции для главного экрана
 * Источник: iOS FunctionCard.swift
 */

enum class StatusType(val indicator: String, val color: Color) {
    ACTIVE("🟢", SuccessGreen),
    WARNING("⚠️", WarningOrange),
    INACTIVE("🔴", DangerRed),
    NEUTRAL("✅", PrimaryBlue)
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun FunctionCard(
    icon: String,
    title: String,
    subtitle: String,
    status: StatusType = StatusType.NEUTRAL,
    modifier: Modifier = Modifier,
    onClick: () -> Unit = {}
) {
    Card(
        onClick = onClick,
        modifier = modifier
            .fillMaxWidth()
            .height(Size.FunctionCardHeight),
        shape = RoundedCornerShape(CornerRadius.Large),
        colors = CardDefaults.cardColors(
            containerColor = Color.Transparent
        ),
        elevation = CardDefaults.cardElevation(
            defaultElevation = 4.dp
        )
    ) {
        Box(
            modifier = Modifier
                .fillMaxSize()
                .background(
                    brush = Brush.linearGradient(
                        colors = listOf(
                            GradientMiddle.copy(alpha = 0.7f),
                            GradientEnd.copy(alpha = 0.7f)
                        )
                    )
                )
                .padding(Spacing.CardPadding),
            contentAlignment = Alignment.Center
        ) {
            Column(
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.spacedBy(Spacing.M)
            ) {
                // Иконка
                Text(
                    text = icon,
                    style = MaterialTheme.typography.displayLarge
                )
                
                // Текст
                Column(
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.spacedBy(Spacing.XXS)
                ) {
                    Text(
                        text = title,
                        style = MaterialTheme.typography.displaySmall,
                        color = TextPrimary,
                        textAlign = TextAlign.Center
                    )
                    
                    Text(
                        text = subtitle,
                        style = MaterialTheme.typography.bodySmall,
                        color = TextSecondary,
                        textAlign = TextAlign.Center
                    )
                }
                
                // Индикатор статуса
                Row(
                    horizontalArrangement = Arrangement.spacedBy(Spacing.XXS),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Box(
                        modifier = Modifier
                            .size(Size.StatusIndicator)
                            .background(status.color, CircleShape)
                    )
                    
                    Text(
                        text = status.indicator,
                        style = MaterialTheme.typography.bodySmall
                    )
                }
            }
        }
    }
}

File: ALADDIN_NEW/mobile_apps/ALADDIN_iOS/Shared/Components/Cards/FunctionCard.swift
import SwiftUI

/// 📇 Function Card
/// Карточка функции для главного экрана (Семья, Защита, Аналитика, AI)
/// Источник дизайна: HTML .function-card на 01_main_screen.html
struct FunctionCard: View {
    
    // MARK: - Properties
    
    let icon: String
    let title: String
    let subtitle: String
    let status: StatusType
    let action: () -> Void
    
    // MARK: - Status Type
    
    enum StatusType {
        case active     // 🟢 Активно
        case warning    // ⚠️ Предупреждение
        case inactive   // 🔴 Неактивно
        case neutral    // ⚪ Нейтрально
        
        var indicator: String {
            switch self {
            case .active: return "🟢"
            case .warning: return "⚠️"
            case .inactive: return "🔴"
            case .neutral: return "✅"
            }
        }
        
        var color: Color {
            switch self {
            case .active: return .successGreen
            case .warning: return .warningOrange
            case .inactive: return .dangerRed
            case .neutral: return .primaryBlue
            }
        }
    }
    
    // MARK: - Init
    
    init(
        icon: String,
        title: String,
        subtitle: String,
        status: StatusType = .neutral,
        action: @escaping () -> Void
    ) {
        self.icon = icon
        self.title = title
        self.subtitle = subtitle
        self.status = status
        self.action = action
    }
    
    // MARK: - Body
    
    var body: some View {
        Button(action: {
            let generator = UIImpactFeedbackGenerator(style: .medium)
            generator.impactOccurred()
            action()
        }) {
            VStack(spacing: Spacing.m) {
                // Иконка
                Text(icon)
                    .font(.system(size: Size.iconLarge))
                
                // Текст
                VStack(spacing: Spacing.xxs) {
                    Text(title)
                        .font(.h3)
                        .foregroundColor(.textPrimary)
                    
                    Text(subtitle)
                        .font(.caption)
                        .foregroundColor(.textSecondary)
                        .multilineTextAlignment(.center)
                }
                
                // Индикатор статуса
                HStack(spacing: Spacing.xxs) {
                    Circle()
                        .fill(status.color)
                        .frame(width: Size.statusIndicator, height: Size.statusIndicator)
                    
                    Text(status.indicator)
                        .font(.caption)
                }
            }
            .frame(maxWidth: .infinity)
            .frame(height: Size.functionCardHeight)
            .padding(Spacing.cardPadding)
            .background(
                // Градиент как в HTML
                LinearGradient.cardGradient
                    .overlay(
                        // Glassmorphism эффект
                        Color.white.opacity(0.05)
                    )
            )
            .cornerRadius(CornerRadius.large)
            .overlay(
                RoundedRectangle(cornerRadius: CornerRadius.large)
                    .stroke(Color.white.opacity(0.1), lineWidth: 1)
            )
            .cardShadow()
        }
        .buttonStyle(PlainButtonStyle())
    }
}

// MARK: - Preview

#Preview {
    VStack(spacing: Spacing.m) {
        // Активная карточка
        FunctionCard(
            icon: "👨‍👩‍👧‍👦",
            title: "СЕМЬЯ",
            subtitle: "4 члена • Всё под защитой",
            status: .active
        ) {
            print("Открыть семью")
        }
        
        // Предупреждение
        FunctionCard(
            icon: "📊",
            title: "АНАЛИТИКА",
            subtitle: "47 угроз заблокировано",
            status: .warning
        ) {
            print("Открыть аналитику")
        }
        
        // Нейтральная
        FunctionCard(
            icon: "🤖",
            title: "AI ПОМОЩНИК",
            subtitle: "Всегда готов помочь",
            status: .neutral
        ) {
            print("Открыть AI")
        }
    }
    .padding()
    .background(
        LinearGradient.backgroundGradient
    )
}



